File: native_app/communication/MessageKinds.go
package communication

type MessageKind string

const (
	SaveToPath  MessageKind = "SAVE_TO_PATH"
	GetCodeTree MessageKind = "GET_CODE_TREE"
	GetContents MessageKind = "GET_CONTENTS"
	// 他のメッセージ種別をここに追加
)

type SaveToPathRequest struct {
	Kind     MessageKind `json:"kind"`
	FilePath string      `json:"filePath"`
	Content  string      `json:"content"`
}

type SaveToPathResponse struct {
	Status string `json:"status"`
	Error  string `json:"error,omitempty"`
}

type GetCodeTreeRequest struct {
	Kind       MessageKind `json:"kind"`
	TargetPath string      `json:"target_path"`
}

type GetCodeTreeResponse struct {
	Repositories []string `json:"repositories"`
	Error        string   `json:"error,omitempty"`
}

type GetContentsRequest struct {
	Kind      MessageKind `json:"kind"`
	FilePaths []string    `json:"filePaths"`
}

type GetContentsResponse struct {
	Contents []string `json:"contents"`
	Error    string   `json:"error,omitempty"`
}

File: pages/options/src/components/communication/MessageKinds.ts
export interface SaveToPathRequest {
    kind: 'SAVE_TO_PATH';
    filePath: string;
    content: string;
}

export interface SaveToPathResponse {
    status: 'success' | 'fail';
    error?: string;
}

export interface GetCodeTreeRequest {
    kind: 'GET_CODE_TREE';
    target_path: string;
}

export interface GetCodeTreeResponse {
    repositories: string[];
    error?: string;
}

export interface GetContentsRequest {
    kind: 'GET_CONTENTS';
    filePaths: string[];
}

export interface GetContentsResponse {
    contents: string[];
    error?: string;
}